test/Npgsql.Tests/SchemaTests.cs

```diff
@@ -271,328 +284,231 @@ public async Task Precision_and_scale()
-        }
-        finally
-        {
-            conn.ExecuteNonQuery("DROP TABLE IF EXISTS data");
-        }
+        await using var conn = await OpenConnectionAsync();
+        await using var _ = await CreateTempTable(conn, "f1 INT, f2 INT, UNIQuE (f1, f2)", out var table);
+
+        var dataTable = await GetSchema(conn, "CONSTRAINTCOLUMNS", new[] { null, null, table });
+        var rows = dataTable.Rows.Cast<DataRow>().ToList();
     }
 
     [Test]
     public async Task Unique_index_composite()
     {
-        using var conn = OpenConnection();
-        try
-        {
-            conn.ExecuteNonQuery(@"
-DROP TABLE IF EXISTS data;
-CREATE TABLE data (f1 INT, f2 INT);
-CREATE UNIQUE INDEX idx_unique ON data (f1, f2);
-");
-            var database = conn.ExecuteScalar("SELECT current_database()");
-
-            string[] restrictions = { null!, null!, "data" };
-            var dataTable = await GetSchema(conn, "INDEXES", restrictions);
-            var index = dataTable.Rows.Cast<DataRow>().Single();
-
-            Assert.That(index["table_schema"], Is.EqualTo("public"));
-            Assert.That(index["table_name"], Is.EqualTo("data"));
-            Assert.That(index["index_name"], Is.EqualTo("idx_unique"));
-            Assert.That(index["type_desc"], Is.EqualTo(""));
-
-            string[] indexColumnRestrictions = { null!, null!, "data" };
-            var dataTable2 = await GetSchema(conn, "INDEXCOLUMNS", indexColumnRestrictions);
-            var columns = dataTable2.Rows.Cast<DataRow>().ToList();
-
-            Assert.That(columns.All(r => r["constraint_catalog"].Equals(database)));
-            Assert.That(columns.All(r => r["constraint_schema"].Equals("public")));
-            Assert.That(columns.All(r => r["constraint_name"].Equals("idx_unique")));
-            Assert.That(columns.All(r => r["table_catalog"].Equals(database)));
-            Assert.That(columns.All(r => r["table_schema"].Equals("public")));
-            Assert.That(columns.All(r => r["table_name"].Equals("data")));
-            Assert.That(columns.All(r => r["index_name"].Equals("idx_unique")));
-
-            Assert.That(columns[0]["column_name"], Is.EqualTo("f1"));
-            Assert.That(columns[1]["column_name"], Is.EqualTo("f2"));
-
-            string[] indexColumnRestrictions3 = { (string) database! , "public", "data", "idx_unique", "f1" };
-            var dataTable3 = await GetSchema(conn, "INDEXCOLUMNS", indexColumnRestrictions3);
-            var columns3 = dataTable3.Rows.Cast<DataRow>().ToList();
-            Assert.That(columns3.Count, Is.EqualTo(1));
-            Assert.That(columns3.Single()["column_name"], Is.EqualTo("f1"));
-        }
-        finally
-        {
-            conn.ExecuteNonQuery("DROP TABLE IF EXISTS data");
-        }
+        await using var conn = await OpenConnectionAsync();
+        await using var _ = await CreateTempTable(conn, "f1 INT, f2 INT, CONSTRAINT idx_unique UNIQuE (f1, f2)", out var table);
```

Comment:
Aren't constraint's names have to be unique too?

Reply:
Also, `UNIQuE` :D

Reply:
> Aren't constraint's names have to be unique too?

Sure, but do you see any problem here?

Reply:
Any other test which attempts to create a constraint with the same name will fail.

Reply:
Ah sure... That's pretty unlikely to happen in the tests, but I'll add the logic for generation of unique constraint names.

Reply:
Definitely not unlikely: COPY tests were failing because of that.